rust/main/utils/run-locally/src/radix/cli.rs

```diff
@@ -0,0 +1,541 @@
+use crate::{
+    logging::log,
+    radix::{types::Contracts, KEY},
+};
+use core_api_client::models::{self, StateUpdates};
+use ethers_core::rand;
+use hyperlane_core::{H256, H512};
+use hyperlane_radix::RadixGatewayProvider;
+use hyperlane_radix::RadixProvider;
+use radix_common::prelude::*;
+use radix_engine_interface::metadata_init;
+use radix_transactions::{
+    builder::ManifestBuilder,
+    model::{
+        HasTransactionIntentHash, PreparationSettings, TransactionHashBech32Encoder,
+        TransactionHeaderV1, TransactionPayload,
+    },
+    prelude::{TransactionBuilder, TransactionManifestV1Builder},
+};
+use scrypto::{
+    blueprints::{package::PackageDefinition, resource::OwnerRole},
+    prelude::dec,
+};
+use std::collections::HashMap;
+use std::{fs, ops::Deref, path::Path, time::Duration};
+use tokio::time::sleep;
+
+impl Deref for RadixCli {
+    type Target = RadixProvider;
+
+    fn deref(&self) -> &Self::Target {
+        &self.provider
+    }
+}
+
+pub struct RadixCli {
+    provider: RadixProvider,
+    package: Option<PackageAddress>,
+    signer: ComponentAddress,
+    network: NetworkDefinition,
+    decoder: AddressBech32Decoder,
+    encoder: AddressBech32Encoder,
+}
+
+impl Clone for RadixCli {
+    fn clone(&self) -> Self {
+        Self {
+            provider: self.provider.clone(),
+            package: self.package.clone(),
+            signer: self.signer.clone(),
+            network: self.network.clone(),
+            decoder: AddressBech32Decoder::new(&self.network),
+            encoder: AddressBech32Encoder::new(&self.network),
+        }
+    }
+}
+
+impl RadixCli {
+    pub fn new(provider: RadixProvider, network: NetworkDefinition) -> RadixCli {
+        let decoder = AddressBech32Decoder::new(&network);
+        let encoder = AddressBech32Encoder::new(&network);
+        let private_key = provider.get_signer().unwrap().get_signer().unwrap();
+        let pub_key = private_key.public_key();
+        let account = ComponentAddress::preallocated_account_from_public_key(&pub_key);
+
+        Self {
+            provider,
+            package: None,
+            signer: account,
+            decoder,
+            encoder,
+            network,
+        }
```

Comment:
It is not a problem from compiler perspective, but let's order fields as they defined in the struct.